contracts/src/Functions.sol

```diff
@@ -65,7 +66,13 @@ library Functions {
             revert InvalidAmount();
         }
 
+        uint256 balanceBefore = IERC20(token).balanceOf(agent);
         IERC20(token).safeTransferFrom(sender, agent, amount);
+
+        uint256 balanceAfter = IERC20(token).balanceOf(agent);
+        if (balanceAfter != balanceBefore + amount) {
+            revert TransferAmountMismatch();
```

Comment:
simplification:
```suggestion
        if (IERC20(token).balanceOf(agent) != balanceBefore + amount) {
            // Tokens with Fee-On-Transfer behaviour are not supported
            revert InvalidToken();
```

Reply:
Resolved in [f3b4caa](https://github.com/Snowfork/snowbridge/pull/1636/commits/f3b4caa7d36be5f4f01c23979a55b433a6802b33)